### DISCUSSION TITLE
Using multiple Netbox frontends with a single PostgreSQL database

### DISCUSSION THREAD

**Discussion by vladimirs-git** [netbox-community/netbox#14413] [Help Wanted!] [answered]
I need to run 2 Netboxes that are connected to the same PostgreSQL DB (single source of truth).
The first is used as a common stable interface, while the second one is used for testing new features.

I have installed a similar setup on my laptop and it is working (same DATABASE, SECRET_KEY settings in configuration.py).
However, before launching this setup in production, I would like to get feedback from experienced who have used a similar solution.

I am concerned about the stability of this solution in write mode.
Each Netbox instance has independent Redis caches, but I am unsure about what would happen if both instances try to update the same objects in DB.
Is possible state, when Netboxes display different information over a long period of time?

**Comment by candlerb** [comment, marked as answer]
> Each Netbox instance has independent Redis caches, but I am unsure about what would happen if both instances try to update the same objects in DB.

Sharing the postgres database absolutely fine. Indeed, this is how it works in single-server operation, where you have multiple gunicorn worker processes and/or threads.

When doing this, you should share the Redis cache as well.

However, the fundamental problem you will have is that there are many cases where the database schema changes between versions. Therefore in general you can't test a new version alongside running an old version using the same database.

The solution is to [clone](https://docs.netbox.dev/en/stable/administration/replicating-netbox/) the production database, copy it to the test server, and then apply the schema changes there (run `./upgrade.sh`).  Of course, any *data* changes made to the test server will not appear in the production server.

For patch versions (e.g. 3.6.1 -> 3.6.2) often there is no schema change, in which case it will be fine to run both in parallel, but in my experience such changes are primarily bugfixes and rarely cause problems, so the benefit of such testing is minimal.

**Reply by vladimirs-git** [reply]
Thanks for your experience!
